ReviewChecklist.md

```diff
@@ -0,0 +1,153 @@
+    - [ ] **Is the option listed for every input mode in [`CommandLineParser` tests](test/solc/CommandLineParser.cpp)?**
+- [ ] **Did you consider interactions with other language features?**
+    - [ ] Are all types covered? Structs? Enums? Contracts/libraries/interfaces? User-defined value types?
+        Values types: integers, fixed bytes, `address`, `address payable`, `bool`? Function pointers?
+        Static and dynamic arrays? `string` and `bytes`? Mappings?
+        Values of that can't be named: literals, tuples, array slices, storage references?
+    - [ ] If it accepts a function, does it also accept an event or an error? These have function types.
+    - [ ] If it affects free functions, what about internal library functions?
+    - [ ] Bound library functions? Functions bound with `using for`?
+    - [ ] Possible combinations of `storage`, `memory`, `calldata`, `immutable`, `constant`?
+        Remember that internal functions can take `storage` arguments.
+    - [ ] Does it work at construction time as well? What if you store it at construction time and read after deployment?
+    - [ ] What about importing it from a different module or inheriting it?
+    - [ ] Have you tested the ternary operator?
+
+### Test style and good practices
+- [ ] **Make test case file names long and specific enough** that it's easy to identify them later.
+    When checking if we have the case already covered the name is usually the only clue we see.
+    - [ ] Place them in the right subdirectory.
+    - [ ] **Avoid simply appending numbers to the name to distinguish similar cases.**
+        Coming up with good names is hard but figuring out if any of hundreds of tests with names that
+        match your search actually fits your case is even harder.
+- [ ] **Don't include version pragma and the SPDX comment in semantic and syntax test cases**.
+    In other test types include them if necessary to suppress warnings.
+- [ ] **If you have to use a version pragma, avoid hard-coding version.** Use `pragma solidity *`.
+- [ ] **Add `--pretty-print --pretty-json 4` to the `args` file of in command-line tests** to get
+    readable, indented output.
+- [ ] **When writing StandardJSON command-line tests, use `urls` instead of `content`** and put
+    the Solidity or Yul code in a separate file.
+
+### Compiler core
+- [ ] **Are error messages sensible and understandable to users?**
+- [ ] **Are error codes consistent?**
+    - [ ] Avoid randomly changing or reassigning error codes.
+    - [ ] Avoid defining separate codes for trivial variants of the same issue.
+        Make it easy for tools to consistently refer to the same error with the same code.
+- [ ] **Prefer Ranges v3 to Boost where possible.**
+- [ ] **Error messages should end with a full stop.**
+
+## Final checks before merging
+- [ ] **Is the PR rebased on top of the `develop` branch** (or `breaking` if it's a breaking change)?
+- [ ] **Did all CI checks pass?**
```

Comment:
I think it's worth mentioning that external tests may fail from time to time, and that most of them are not required steps for merging.

Reply:
Done.